sparse_strips/vello_common/src/glyph.rs

```diff
@@ -140,20 +134,30 @@ impl<'a, T: GlyphRenderer + 'a> GlyphRunBuilder<'a, T> {
                 continue;
             }
 
-            let mut local_transform =
-                Affine::translate(Vec2::new(glyph.x as f64 * scale, glyph.y as f64 * scale));
-            if let Some(skew) = horizontal_skew {
-                local_transform *= Affine::skew(skew.tan() as f64, 0.0);
-            }
-            if let Some(glyph_transform) = glyph_transform {
-                local_transform *= glyph_transform;
+            // Calculate the global glyph translation based on the glyph's local position within
+            // the run and the run's global transform.
+            //
+            // This is a partial affine matrix multiplication, calculating only the translation
+            // component that we need. It is added below to calculate the total transform of this
+            // glyph.
+            let [a, b, c, d, _, _] = self.run.transform.as_coeffs();
+            let translation = Vec2::new(
+                a * glyph.x as f64 + c * glyph.y as f64,
+                b * glyph.x as f64 + d * glyph.y as f64,
+            );
+
+            // When hinting, ensure the y-offset is integer. The x-offset doesn't matter, as we
+            // perform vertical-only hinting.
+            let mut total_transform = transform.then_translate(translation).as_coeffs();
```

Comment:
This feels suspicious to me. In theory, we want to skew, then translate locally, then transform globally, right?
But this ordering seems to me to be translate locally, then skew, then transform globally.

In any case, we need a test which has a long-ish text run skewed using `glyph_transform`.

Reply:
For posterity: we apply `glyph_transform`, then translate with the glyph position from shaping/layouting, then apply the global `transform`.

Mathematically that's `transform * glyph_translation * glyph_transform`, where all of these are 3x3 affine matrices.

`glyph_translation` only adds a translation. That means we can equivalently calculate `transform * glyph_transform`, and then add the (transformed) translation component from `glyph_translation` to the total transform. This order is slightly cheaper to calculate.